src/modm/platform/core/avr/assert.cpp

```diff
@@ -46,7 +46,8 @@ void modm_assert_fail_context(const char * identifier, uintptr_t context)
 		state & (uint8_t) Abandonment::Fail)
 	{
 		modm_abandon(module, location, failure, context);
-		while(1) ;
+		while (true)
+			{};
```

Comment:
No. Ugly.

Reply:
[MISRA Rule 59](https://lost-contact.mit.edu/afs/rhic.bnl.gov/i386_sl305/app/codewizard-4.3/manuals/items_ne.htm#1020987) This rule / [suggestion](https://www.perforce.com/resources/qac/misra-c-cpp) is to avoid errors when the code is expanded.
What would you suggest? One-liner?

Reply:
Just like the Law™ isn‘t a good moral guide, MISRA is not a good style guide.
Besides, what‘s the point of *selectively* quoting some random MISRA rule here? We‘re not using MISRA as a style guide.

I don‘t see a good argument for any of these changes. It‘s basically that you like it more that way. Well, I don‘t like this style, but I can live with these changes.